pytorch_lightning/loops/base.py

```diff
@@ -45,9 +47,36 @@ class Loop(ABC):
 
     def __init__(self) -> None:
         self.iteration_count: int = 0
-        self.trainer: Optional['pl.Trainer'] = None
+        self._trainer: Optional['pl.Trainer'] = None
         self.restarting = False
 
+    @property
+    def loop_progress(self) -> Dict[str, Any]:
```

Comment:
this is for internal use, right? so we can make it protected and distinguish from the other progress attribute? Otherwise it will be confusing that there are two public properties for progress